app/models/package_manager/pypi/json_api_project.rb

```diff
@@ -137,7 +137,7 @@ def deprecation_status
         # on Releases. Make sure that the code here and in the VersionProcessor
         # do not conflict.
         # TODO: refactor this deprecation method to reuse the VersionProcessor status logic
-        if stable_releases.all?(&:yanked?)
```

Comment:
I think another option is to consider every release instead of just stable ones and pull the reason from the most recent release even if it's a pre-release version number.